packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -14,21 +14,22 @@ import StoreKit
 public class InAppPurchasePlugin: NSObject, FlutterPlugin, InAppPurchaseAPI {
   private let receiptManager: FIAPReceiptManager
   private var productsCache: NSMutableDictionary = [:]
-  private var paymentQueueDelegateCallbackChannel: FlutterMethodChannel?
+  private var paymentQueueDelegateCallbackChannel: MethodChannel?
   // note - the type should be FIAPPaymentQueueDelegate, but this is only available >= iOS 13,
   // FIAPPaymentQueueDelegate only gets set/used in registerPaymentQueueDelegateWithError or removePaymentQueueDelegateWithError, which both are ios13+ only
   private var paymentQueueDelegate: Any?
-  private var requestHandlers = Set<FIAPRequestHandler>()
-  private var handlerFactory: ((SKRequest) -> FIAPRequestHandler)
+  // Sets do not accept protocols
```

Comment:
what do you mean by "sets do not accept protocols"? what were you trying to write? and what error did you get?

Reply:
swift sets cant contain protocols
eg. Cant have Set<PaymentQueue>, has to be a concrete type
https://stackoverflow.com/questions/33189345/set-and-protocols-in-swift

Reply:
you will have to type erase the protocol in order to use it in set. something like https://medium.com/@elifilaykandemir/type-erasure-in-swift-04a76ec47503